ddl/index.go

```diff
@@ -599,14 +599,27 @@ func doReorgWorkForCreateIndex(w *worker, d *ddlCtx, t *meta.Meta, job *model.Jo
 	// environment will be set up and then use reorgInfo.IsLightningok to show whether the lightning
 	// backfill used for specfic ddl job.
 	if isAllowFastDDL(reorgInfo.d.store) {
-		// init the flag, set IsLightningOk to false first.
-		reorgInfo.IsLightningEnabled = false
-		// If get tblInfo failed, go back to kernel backfill process.
-		if err == nil {
-			err = prepareBackend(w.ctx, indexInfo.Unique, job, reorgInfo.ReorgMeta.SQLMode)
-			// Once Env is created well, set IsLightningOk to true.
+		// Check if the reorg task is re-entry task, If TiDB is restarted, then currently
+		// reorg task should be restart.
+		if !canRestoreReorgTask(reorgInfo, indexInfo.ID) {
+			// Init the flag, set IsLightningOk to false first.
+			reorgInfo.IsLightningEnabled = false
+			// If get tblInfo failed, go back to kernel backfill process.
 			if err == nil {
-				reorgInfo.IsLightningEnabled = true
+				err = prepareBackend(w.ctx, indexInfo.Unique, job, reorgInfo.ReorgMeta.SQLMode)
```

Comment:
canRestoreReorgTask is both check backend and engine;
but if there is re-entry task, it is only prepare backend but not prepare engine , it is enough??